stdgo/reflect/Reflect.hx

```diff
@@ -462,17 +477,44 @@ class Type {
 	public function numMethod():GoInt {
 		switch (gt) {
 			case GT_namedType(_,_,methods,_,_,_), GT_interface(_,_,methods):
-				return methods.length; //TODO
+				return methods.length;
 			default:
 				throw new Error("reflect.NumMethod not implemented for " + toString());
 		}
 		return 0;
 	}
 
+	public function field(index:GoInt):StructField {
+		switch gt {
+			case GT_namedType(module, name, methods, fields, interfaces, type):
+				var field = fields[index.toBasic()];
+				switch field {
+					case GT_field(name, type, tag):
+						return {
+							name: name,
+							pkgPath: "",
+							type: new Type(type),
+							tag: tag,
+							offset: 0,
+							index: null,
```

Comment:
`index` is a `[]int` of indexes into fields ( a slice to allow nested structures ).

As you already have an `index` to the field passed in, you could create a slice with that single index in it, which would probably be correct for un-nested structures.

Reply:
Thank you these tips are extremly helpful, I'll make the change for un-nested structures, not sure how to add nested structure types. I'm working on readding your functionality again so apologies if I'm missing obvious things.